src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -740,7 +716,11 @@ namespace winrt::TerminalApp::implementation
     //   when this is called, nothing happens. See _ShowDialog for details
     winrt::Windows::Foundation::IAsyncOperation<ContentDialogResult> TerminalPage::_ShowQuitDialog()
     {
-        return _ShowDialogHelper(L"QuitDialog");
+        if (auto presenter{ _dialogPresenter.get() })
+        {
+            co_return co_await presenter.ShowDialog(FindName(L"QuitDialog").try_as<WUX::Controls::ContentDialog>());
+        }
+        co_return ContentDialogResult::None;
```

Comment:
Why is the `_ShowDialogHelper` gone? It was abstracting away this redundant code.
If you keep `_ShowDialogHelper`, you only need to remove 3 lines instead of adding 20. 🤔

Reply:
Oh that's probably a good idea. I just blindly reverted the two commits, forgot there was goodness in there besides the fix.